### DISCUSSION TITLE
How to optimize generated file size

### DISCUSSION THREAD

**Discussion by msickpaler** [dotansimha/graphql-code-generator#10060] [Q&A]
# Large generated file size with GraphQL Code Generator

## Environment

- Next.js: 14.2.3
- @graphql-codegen/cli: 5.0.2

## Description

I'm experiencing an issue with the file size of the generated code from GraphQL Code Generator. The generated files are becoming excessively large, which is impacting our project's performance and build times.

## Current file size

5.8MiB

## Files

```ts
// codegen.ts
const config = {
  overwrite: true,
  schema: "https://postgraphile.example/graphql",
  documents: ["src/graphql/**/*.graphql"],
  generates: {
    "src/graphql/generated/": {
      preset: "client",
      config: {
        onlyOperationTypes: true,
        enumsAsTypes: true,
      },
    },
  },
};

export default config;
```

## Expected Behavior

The generated files should be more optimized and smaller in size.

## Actual Behavior

The generated files are significantly large, causing issues when locally developing.

## Additional Context

- We have a very large number of tables in our database(about 300), and this number is expected to increase further.
- We tried to use the swc plugin (https://the-guild.dev/graphql/codegen/plugins/presets/preset-client#reducing-bundle-size), but the swc version is not compatible with Next.js 14.2.3.
- While we understand that Near Operation File Preset could be used to split the operation-related code, our main concern is the large number of tables. We're looking for a way to split the type definitions themselves.
- We're considering options to split the type definitions, as the table count is the primary issue causing the large file size.

Any assistance or insights on how to address this issue, particularly regarding splitting type definitions for a large number of tables, would be greatly appreciated. Thank you!

**Comment by l-you** [comment]
@dotansimha any idea?
